Errors in 4.14.1

## Issue Details

The 4.14.1 testsuite shows several problems : 

- [x]  [in Intersections_2](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14.1-I-173/Intersections_2/TestReport_cgaltester_x86-64_Darwin-18.2_Apple-clang-default_Release.gz) (fixed in #4037, merged)

- [x] [in mesh_3_Examples](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14.1-I-173/Mesh_3_Examples/TestReport_lrineau_CentOS6-32.gz) (fixed in #4038, merged)

- [x] apparently there is some c++11 in [Triangulation_2 ](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14.1-I-173.shtml#Triangulation_2) (fixed in #4036, merged)

- [x] [in the Classification_plugin of the demo](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14.1-I-173/Polyhedron_Demo/TestReport_gimeno_CentOS7.gz) (fixed in #4048, merged)

- [x] there are some new uninitialized variables in [Kernel_23](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14.1-I-173.shtml#Kernel_23) (fixed by #4040, merged)

- [x] <s>and some warnings in [Generator](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14.1-I-173.shtml#Generator)</s>(have been there in CGAL since early 2018: that depends on the Boost version)


- [x] Also, recent testsuites have shown problemes on Gauguin : 
In debug `std::byte` is said to be ambiguous, and in Release mode the Polyhedorn demo is drowning in boost errors. (fixed by #4034, merged)

## Environment

* CGAL version:4.14.1

As for the compilation error in Classsification plugin, I let @sgiraudot deal with it once he comes back.
```
/home/cgal_tester/build/src/cmake/platforms/CentOS7/test/Polyhedron_Demo/Plugins/Classification/Cluster_classification.cpp:942:50:   required from here
/usr/include/boost/utility/result_of.hpp:166:8: error: no class template named 'result' in 'const struct CGAL::Classification::Cluster<ItemRange, ItemMap>::bbox() const [with ItemRange = Point_set_3<CGAL::Mesh_3::Robust_intersection_traits_3<CGAL::Epick> >; ItemMap = CGAL::Point_set_3<CGAL::Point_3<CGAL::Epick>, CGAL::Vector_3<CGAL::Epick> >::Property_map<CGAL::Point_3<CGAL::Epick> >]::__lambda1'
 struct result_of_nested_result : F::template result<FArgs>
        ^
```
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.14.1-I-173/Polyhedron_Demo/TestReport_gimeno_CentOS7.gz

It is very recent. Probably because of https://github.com/CGAL/cgal/pull/3961.

--- a/Triangulation_2/test/Triangulation_2/issue_4010.cpp
+++ b/Triangulation_2/test/Triangulation_2/issue_4010.cpp
@@ -4,33 +4,32 @@
 #include <CGAL/Constrained_triangulation_plus_2.h>
 
 
-typedef CGAL::Exact_predicates_inexact_constructions_kernel                 K;
+typedef CGAL::Exact_predicates_inexact_constructions_kernel               K;
 typedef CGAL::Polygon_2<K>                                                Polygon_2;
 typedef CGAL::Exact_intersections_tag                                     Itag_;
 typedef CGAL::Constrained_Delaunay_triangulation_2<K,CGAL::Default, Itag_> CDT;
 typedef CGAL::Constrained_triangulation_plus_2<CDT>                       CDTP;
 
 typedef CDTP::Point                                                       Point;
-typedef CDTP::Constraint_id                                               Cid;
 typedef CDTP::Vertex_handle                                               Vertex_handle;
-
+typedef CDTP::Constraint_id                                               Constraint_id;
 
 int main()
 {
   CDTP cdtp;
   
-  Vertex_handle handle1=cdtp.insert(Point(0,0));
-  Vertex_handle handle2=cdtp.insert(Point(0,10));
-  Vertex_handle handle3=cdtp.insert(Point(10,10));
-  Vertex_handle handle4=cdtp.insert(Point(10,0));
+  Vertex_handle handle1 = cdtp.insert(Point(0,0));
+  cdtp.insert(Point(0,10));
+  Vertex_handle handle3 = cdtp.insert(Point(10,10));
+  cdtp.insert(Point(10,0));
   
-  auto id = cdtp.insert_constraint(handle1,handle3);
+  cdtp.insert_constraint(handle1,handle3);
   
   cdtp.remove_constraint(handle1,handle3);
   
-  auto lastConstraintId=cdtp.insert_constraint(handle1,handle3);
+  auto lastConstraintId = cdtp.insert_constraint(handle1,handle3);
   
-  if (lastConstraintId == nullptr){
+  if (lastConstraintId == Constraint_id()){
     std::cout << "problem" << std::endl;
     assert(false);
   }

